BatchApex/BatchApex.cls

```diff
@@ -0,0 +1,119 @@
+/**
+ * Collection of Closed Won Opportunities 
+ * The Primary contact associated with an account needs to know which all are the 
+ * closed won opportunities for his account per day. 
+ * Create a process which will run everyday in the midnight & collect all the 
+ * Closed won opportunities of that account in a day & send an email to Primary Contact. 
+ * The email body should contain the Opportunity Name & it's respective amount 
+ * with the aggregated amount in the end. 
+ */
+global with sharing class BatchApex implements Database.Batchable<Sobject>, Database.Stateful, 
+    Schedulable {
+    map<String,WrapperClass> accountOpportunityMap;
+    public BatchApex() {
+        accountOpportunityMap = new Map<String,WrapperClass>();
+    }
+    
+    /**
+     * Wrapper class with opportunity and sum
+     */
+    public class WrapperClass {
+         List<Opportunity> opportunityList;
+         Double sum;
+         
+        /**
+         *Constructor to initialize the warpper class variables
+         */
+        public WrapperClass() {
+            opportunityList = new List<Opportunity>();
+            sum = 0;
+        }
+    }
+    
+    /**
+     * Get all opportunities whos lookup field to account is not blank 
+     * and email field on Account object is not blank
+     */
+    global Database.QueryLocator start(Database.BatchableContext batchableContext) {
+            return Database.getQueryLocator([
```

Comment:
This query will return all the opportunity records every day. But you  assignment says - "The Primary contact associated with an account needs to know which all are the closed won opportunities for his account per day." i.e opportunites that we create per day should be sent on per day basis. But in thiscase everyday all the opportunity records will be sent. Please chnage this. Let me know if u have any doubts or anything not clear